### ISSUE TITLE
Updated to new Mealie version yesterday and account/recipe add-on data is now unavailable

### ISSUE THREAD

**Issue by SeeThisIsMe** [alexbelgium/hassio-addons#371]
Hi,

There was a new Mealie add-on version released for Home Assistant yesterday, and I performed the standard update as prompted in HA (taking a HA backup at the same time – shown in attached screenshot). Following that I could no longer log into Mealie and received a user credential error. I then tried logging in using the default Mealie credentials which worked, but there were no other accounts present in Mealie and no recipe data.

It seems the update has wiped out a lot of the add-on data in the process, so I restored the backup that was taken using HA at the time of the update, but that does not restore my user account or the recipe data in Mealie - I am still left with what seems like an empty installation of Mealie.

I extracted the backup file and can see some Mealie files and folders contained inside, such as the folder structure and images for the recipes, but none of the JSON files that contain the actual recipe information are present. 

So it seems that something is not working correctly with the update, and with the backups being performed for the add-on (and being that this is a recipe manager add-on I expect all the recipe data should be contained in the HA backups when those are done). 

I’m not sure what info may assist to investigate this, but I’m happy to provide it if you can advise. Also if necessary I am happy to remove and reinstall the add-on as new, but I need to know where I can find all the data for the recipes that were there prior to the update.

Thanks.
![backup](https://user-images.githubusercontent.com/53046241/175224247-b6839f10-d517-408e-b19c-5dffe4126f17.JPG)

**Comment by alexbelgium** [comment]
Hello, sorry about that. All data is stored in the addon itself through the sqlite database, so all your data should be safe.

Which architecture are you using? Amd64? Aarch64? Thank you

**Comment by SeeThisIsMe** [comment]
OK great. It's Amd64. Many thanks.

**Comment by alexbelgium** [comment]
Ah, that's why! Amd64 is upgrade to the greatest and newest v1.0, but indeed I will have to check if the app dev has taken into account database upgrade.

On the other hand I'm surprised that the backup doesn't restore, it should work... I'll do some tests but whatever you do keep your backup intact ;)

**Comment by alexbelgium** [comment]
Oh joy - from the dev website : "Currently Mealie provides no database migrations as doing so would slow down development and hinder major changes that may need to happen prior to v1.0.0. Mealie has a robust backup and restore system for managing your data."

Well, that means we must focus on bringing your backup online again, from which you will be able to perform a database backup, then upgrade, then restore your recipes.

Could you please send me the full log that you see when you restore your backup ? and do you confirm that when restoring your backup it is still your "old" username & password? Thank you

**Comment by SeeThisIsMe** [comment]
Ahh... OK. I have an idea that could be quicker and easier. My HA instance is running as a VM on an ESXi host server. So I could quickly try spinning up another VM and install HA onto that, and then restore the HA Mealie add-on backup there. If that works I could then do an account/recipe backup from within Mealie and then import that back into my original HA/Mealie instance. 

I'll let you know how it goes. You might however want to flag this new version with the lack of Mealie database migration support on the release notes for the add-on - saying it will need a backup/restore to be done within Mealie itself or else data will be lost. Might save some other people from the same pain :)

Thanks.

**Comment by alexbelgium** [comment]
Good idea.

For flagging people actually there is no real way to do that in HA... I had written a note in the changelog but people who like me have automatic updating enabled don't read the changelogs. I'll put a note on top of Readme, that should make people more aware of this :-) thanks

**Comment by SeeThisIsMe** [comment]
So I did the process I described and it worked like a charm :) All the recipes are there. 

Something to note however about the database migration is that only the recipe data gets migrated. Not user data or other settings, and the favorite recipes are also no longer listed as such so they need to be selected again. 

Thanks for your help! Now I'll get a regular scheduled recipe backup sorted out ;)
